tests/unittests/decorators.py

```diff
@@ -43,6 +45,27 @@ async def wrapper(*args, **kwargs):
     return wrapper
 
 
+def drop_stores(func, pattern="*"):
```

Comment:
Danger! ☠️☠️☠️

Looks like the default `@drop_stores` syntax will drop all of my persistent dataset's stores!

Note that the existing `@drop_datasets` decorator used in tests only deletes non-persistent datasets, which by definition should be safe to delete.

I definitely think we need to tweak this PR so that running unit tests doesn't affect any persistent datasets that the user may have in their DB.

Reply:
What if we assert here that the dataset is non persistent?

Reply:
Best idea I have so far is to only ever delete stores with a `__test__` prefix - via `@drop_stores(pattern="__test__*")`.

Then I'll rename all the tests store names to match. However this won't solve the issue where tests would fail...that would require some additional isolation. Maybe something like:

```python
original_collection_name = ExecutionStoreService.collection_name
ExecutionStoreService.collection_name = "test_stores"
func(*args, **kwargs*)
ExecutionStoreService.collection_name = original_collection_name
```